src/java/org/apache/cassandra/service/accord/StaleReplicas.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.service.accord;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.annotation.concurrent.Immutable;
+
+import com.google.common.collect.ImmutableSet;
+
+import accord.local.Node;
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.DataOutputPlus;
+import org.apache.cassandra.service.accord.serializers.TopologySerializers;
+import org.apache.cassandra.tcm.Epoch;
+import org.apache.cassandra.tcm.MetadataValue;
+import org.apache.cassandra.tcm.serialization.MetadataSerializer;
+import org.apache.cassandra.tcm.serialization.Version;
+import org.apache.cassandra.utils.CollectionSerializers;
+
+@Immutable
+public class StaleReplicas implements MetadataValue<StaleReplicas>
+{
+    public static final StaleReplicas EMPTY = new StaleReplicas(ImmutableSet.of(), Epoch.EMPTY);
+
+    private final Set<Node.Id> nodeIds;
+    private final Epoch lastModified;
+
+    StaleReplicas(Set<Node.Id> nodeIds, Epoch lastModified)
+    {
+        this.nodeIds = nodeIds;
+        this.lastModified = lastModified;
+    }
+
+    @Override
+    public StaleReplicas withLastModified(Epoch epoch)
+    {
+        return new StaleReplicas(nodeIds, epoch);
+    }
+
+    @Override
+    public Epoch lastModified()
+    {
+        return lastModified;
+    }
+    
+    public StaleReplicas withNodeId(Node.Id nodeId)
+    {
+        Set<Node.Id> copy = new HashSet<>(nodeIds);
```

Comment:
nit, can we make this `ImmutableSet` to avoid having shared objects that have mutable state?